src/contract/event.rs

```diff
@@ -655,6 +652,23 @@ impl<T: Transport, E: ParseLog> AllEventsBuilder<T, E> {
     }
 }
 
+/// Retrieves a block number for the specified transaction hash.
+async fn block_number_from_transaction_hash<T: Transport>(
+    web3: Web3<T>,
+    tx_hash: H256,
+) -> Result<u64, ExecutionError> {
+    let tx_receipt = web3
+        .eth()
+        .transaction_receipt(tx_hash)
+        .compat()
+        .await?
+        .ok_or(ExecutionError::MissingTransaction(tx_hash))?;
+    Ok(tx_receipt
+        .block_number
+        .ok_or(ExecutionError::PendingTransaction(tx_hash))?
+        .as_u64())
+}
+
```

Comment:
Was this moved?

Reply:
Just within the same file up a few lines. Just to move it closer to where it is called.